### DISCUSSION TITLE
Can't find manual controls to play a CD

### DISCUSSION THREAD

**Discussion by subatomicglue** [rern/rAudio#603] [Fixed Issues] [answered]
I tried `Addons > rAudio 1 > long-press Update > ok for UPDATE branch`
But got Download Failed. Got it like 4 times in a row....

I played a web radio thing (which came stock in my 20211022 build - the "rock stream") and it worked, proved internet was connected

Then tried update again, and .... it worked.

Bug?
- Maybe WiFi internet isn't connect until I use it for a stream?   seems like a bug.... maybe....  idk what happened above.
- Now when I insert a CD, I get no playback (and auto play is enabled). Before the update, the CD would play.  ejected and reinserted several times.   UPDATE: after power cycle my DVD/CD player, seems to be back playing tracks from the CD.  Perhaps the update wedged my player...

Feature Request?
- I see no manual controls in "the library" to play tracks on the CD, or to queue songs...   (feature request, if this isn't a bug)
- I dont see the CD player under `Settings / Storage`, but that could be a nice indicator for diagnostics... (feature request)

**Comment by subatomicglue** [comment]
I realize UPDATE probably isn't a stable branch.  Hope the report helps :)

**Comment by subatomicglue** [comment]
Well.     I power cycled my DVD/CD drive....  (I only unplugged USB cable before), and i'm now playing tracks from the CD again.

This may be an issue with my hardware - perhaps.   Or the update wedged my DVD/CD hardware.    

Still, Please read the other items for feature requests.   
- Would be great to have a CD icon under the library, so I can manually go in and out of the CD or queue other music.

**Reply by rern** [reply]
Audio CD tracks:
- On insert - appear on Playlist page which can be dragged/moved or deleted like normal tracks.
- On eject - cleared from Playlist

**Reply by subatomicglue** [reply]
What about:
- On insert - appear on Playlist page which can be dragged/moved or deleted like normal tracks.
- On Playlist queue 'replace' some different audio files,  "replaces" the CD tracks in the queue with some other tracks (e.g. from USB drive)...   (or simply clear the Playlist queue).
- No way to play the inserted audio CD.... unless eject then reinsert

**Reply by subatomicglue** [reply]
Also.....    I have a problem this, which could be causing my confusion and increasing desire for a "play CD" control, as well as desire for a `Settings / System` status diagnostic indicator for the CD drive:    something is quietly failing and I can't tell what's happening:

- On insert - appear on Playlist page which can be dragged/moved or deleted like normal tracks.
- On Playlist queue 'replace' some different audio files (e.g. from USB drive)
- On clear of the Playlist queue.... now there is no way to play the CD  (so I'll remove/insert the CD again... ok)
- On eject of the CD.... 
- On insert of the CD....  nothing happens....   (tried unplugging both power and USB cables to the drive... waiting... I see the blue notification "USB CD Off/Removed" ...  reinsert both.... waiting.... I see the blue notification "USB CD On" ...  but nothing in playlist, no playing audio... then eject... I see the blue notification "USB CD Eject" ... and insert... waiting...  no notification...  nothing in the playlist, no playing audio.).   Playlist Queue stays empty

**Reply by subatomicglue** [reply]
Just upgraded to the recently released 20211112, and this seems a lot more resilient now (maybe even bugfree!) 
- eject/insert seems to always queue CD tracks now
- clearing the playlist - or - replacing the playlist with USB tracks - and then - eject/insert the CD now works for me...

THANK YOU @rern

**Comment by subatomicglue** [comment, marked as answer]
Just upgraded to the recently released 20211112, and this seems a lot more resilient now (maybe even bugfree!)
THANK YOU @rern
